cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocJobMain.java

```diff
@@ -0,0 +1,202 @@
+        closeMethod.invoke(newDataprocEnvInstance);
+        try {
+          // make sure directory is cleaned before exiting
+          deleteDirectoryContents(tempDir.toFile());
+        } catch (IOException e) {
+          LOG.warn("Failed to delete temporary directory {}. ", tempDir, e);
+        }
+      }
+
+      LOG.info("Runtime job completed.");
+    }
+  }
+
+  /**
+   * This method will generate class path by adding following to urls to front of default classpath:
+   *
+   * expanded.resource.jar
+   * expanded.application.jar
+   * expanded.application.jar/lib/*.jar
+   * expanded.application.jar/classes
+   * expanded.twill.jar
+   * expanded.twill.jar/lib/*.jar
+   * expanded.twill.jar/classes
+   *
+   */
+  private static URL[] getClasspath(URLClassLoader cl, File tempDir) throws IOException {
+    URL[] urls = cl.getURLs();
+    List<URL> urlList = new ArrayList<>();
+    for (String file : ImmutableList.of(Constants.Files.RESOURCES_JAR,
+                                        Constants.Files.APPLICATION_JAR,
+                                        Constants.Files.TWILL_JAR)) {
+      File jar = new File(file);
+      File jarDir = new File(tempDir, "expanded." + file);
+      expand(jar, jarDir);
+      // add url for dir
+      urlList.add(jarDir.toURI().toURL());
+      if (file.equals(Constants.Files.RESOURCES_JAR)) {
+        continue;
+      }
+      urlList.addAll(createClassPathURLs(jarDir));
+    }
+
+    urlList.addAll(Arrays.asList(urls));
+
+    LOG.info("Classpath URLs: {}", Arrays.toString(urlList.toArray()));
+    return urlList.toArray(new URL[0]);
+  }
+
+  /**
+   * Expands jar into destination directory.
+   */
+  private static void expand(File jarFile, File destDir) throws IOException {
+    try (ZipInputStream zipIn = new ZipInputStream(new BufferedInputStream(new FileInputStream(jarFile)))) {
+      Path targetPath = destDir.toPath();
+      Files.createDirectories(targetPath);
+
+      ZipEntry entry;
+      while ((entry = zipIn.getNextEntry()) != null) {
+        Path output = targetPath.resolve(entry.getName());
+
+        if (entry.isDirectory()) {
+          Files.createDirectories(output);
+        } else {
+          Files.createDirectories(output.getParent());
+          Files.copy(zipIn, output);
+        }
+      }
+    }
+  }
+
+  private static List<URL> createClassPathURLs(File dir) {
+    try {
+      List<URL> urls = new ArrayList<>();
+      // add jar urls from lib under dir
+      addJarURLs(new File(dir, "lib"), urls);
+      // add classes under dir
+      urls.add(new File(dir, "classes").toURI().toURL());
+      return urls;
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void addJarURLs(File dir, List<URL> result) throws MalformedURLException {
+    Optional<File[]> files = Optional.ofNullable(dir.listFiles(f -> f.getName().endsWith(".jar")));
+    if (files.isPresent()) {
```

Comment:
If you are checking `isPresent`, it will as well be simpler not to use `Optional`, but rather just `if (files != null)`